diff -burN libcfu-0.02.orig/configure.ac libcfu-0.02/configure.ac
--- libcfu-0.02.orig/configure.ac	2005-08-01 09:25:52.000000000 +0000
+++ libcfu-0.02/configure.ac	2005-08-01 09:26:36.000000000 +0000
@@ -8,7 +8,7 @@
 AC_SUBST(LIBCFU_TYPE_u_int)
 
 AC_PROG_CC
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 dnl Check for pthread support
 SAVECFLAGS="$CFLAGS"
--- libcfu-0.03/src/Makefile.am.orig	2024-02-11 10:14:12.098048826 +0100
+++ libcfu-0.03/src/Makefile.am	2024-02-11 10:16:45.943882038 +0100
@@ -1,5 +1,5 @@
-lib_LIBRARIES = libcfu.a
-libcfu_a_SOURCES = cfuhash.c cfutime.c cfustring.c cfulist.c cfuconf.c cfuthread_queue.c cfu.c cfuopt.c
+lib_LTLIBRARIES = libcfu.la
+libcfu_la_SOURCES = cfuhash.c cfutime.c cfustring.c cfulist.c cfuconf.c cfuthread_queue.c cfu.c cfuopt.c
 bin_PROGRAMS = libcfu-config
 
 CFU_CONFIG_DEFINES = -DCFU_LIBDIR="\"@libdir@\"" -DCFU_LIBS="\"@LIBS@\"" -DCFU_INCLUDEDIR="\"@includedir@\"" -DCFU_PREFIX="\"@prefix@\""  -DCFU_EXEC_PREFIX="\"@exec_prefix@\""
@@ -16,13 +16,13 @@ include_HEADERS = $(top_srcdir)/include/
     $(top_srcdir)/include/cfuthread_queue.h $(top_srcdir)/include/cfutypes.h \
     $(top_srcdir)/include/cfuopt.h
 
-# libcfu_a_LDFLAGS = -lpthread
+# libcfu_la_LDFLAGS = -lpthread
 INCLUDES = -I$(top_srcdir)/include
 
-default: libcfu.a libcfu-config
+default: libcfu.la libcfu-config
 
 libcfu-config.o: libcfu-config.c
 
-libcfu-config: libcfu.a libcfu-config.o
-	$(CC) $(CFLAGS) -o libcfu-config libcfu-config.o libcfu.a -pthread
+libcfu-config: libcfu.la libcfu-config.lo
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o libcfu-config libcfu-config.o libcfu.la -pthread
 
